### ISSUE TITLE
Can the status constraint feature property be simplified?

### ISSUE THREAD

**Issue by decentralgabe** [decentralized-identity/presentation-exchange#310]
https://identity.foundation/presentation-exchange/#credential-status-constraint-feature

The spec says:
```
The values of all status properties are objects, composed as follows:

status objects MUST include a directive property, and its value MUST be one of the following strings:
required - the credential MUST be of the specified status.
allowed - the credential MAY be of the specified status.
disallowed - the credential MUST NOT be of the specified status.
  "statuses": {
    "active": {
      "directive": "required"  // other values: "allowed", "disallowed"
    },
    "suspended": {...},
    "revoked": {...}
  }
```

Why objects if they contain a single key and a single value? What about...

```
  "statuses": {
    "active": "allowed",  // other values: "required", "disallowed"
    "suspended": "allowed",
    "revoked": "disallowed",
  }
```

**Comment by OR13** [comment]
Short answer would seem to be that object can be extended.

When writing specs with JSON, prefer objects to arrays or strings, since you can add properties to fix issues that might arise in the future.

**Comment by decentralgabe** [comment]
eh, premature optimization. it's a similar amount of work either way.

**Comment by brentzundel** [comment]
I vaguely recall the conversations that resulted in the value being an object with a `directive` property than simply a value, but I can't recall the exact reasoning why that decision was made. 
I'm fine either way.

**Comment by csuwildcat** [comment]
The reasoning here was that it was likely that you would need to specify which mechanisms you would accept for status checks, not just the actual high-level status value they must be. The addition of which mechanisms are acceptable, with any annotation required to note them, would require at least one additional property. This is admittedly a forward looking structure, but I don't know how else you would do this without allowing for more props as methods for revocation checks become common enough to distinguish between.

**Comment by decentralgabe** [comment]
Discussed on the call today: we will extend the field to add a `type` property a-la StatusList2021

**Comment by decentralgabe** [comment]
https://github.com/decentralized-identity/presentation-exchange/pull/328

**Comment by decentralgabe** [comment]
Fixed in #328
